webapp/test/listitem.spec.js

```diff
@@ -1,56 +1,73 @@
-  });
+let wrapper;
 
-  it('has rows', () =>{
-    expect(wrapper.html()).toContain('<div class="row mx-lg-n5 jest-list-item">');
-  });
-
-  it('has columns', () =>{
-    expect(wrapper.html()).toContain('<div class="col py-3 border bg-light">');
-  });
-
-  it('has a delete button', () =>{
-    expect(wrapper.html()).toContain('id="buttonDelete"');
-    expect(wrapper.html()).toContain('name="delete-button"');
-    expect(wrapper.html()).toContain('type="button"');
-  });
+describe('todo item', () => {
+    beforeEach(() => {
+        wrapper = mount(listitem, {propsData});
+    });
+    it('shows text of todo item', () => {
+        expect(wrapper.contains('#todoText')).toBe(true);
+        expect(wrapper.find('#todoText').text()).toBe(exampleItem.text);
+    });
+    it('shows user name in todo', () => {
+        expect(wrapper.find('#username').text()).toBe(testUserName)
+    });
+    it('has an edit button', () => {
+        expect(wrapper.contains('#buttonEdit')).toBe(true);
+        expect(wrapper.find('#buttonEdit').text()).toBe("Edit");
+    });
+    it('has a delete button', () => {
+        expect(wrapper.contains('#buttonDelete')).toBe(true);
+        expect(wrapper.find('#buttonDelete').text()).toBe("Delete");
+    });
 
-  it('has an edit button', () =>{
-    expect(wrapper.html()).toContain('id="buttonEdit"');
-    expect(wrapper.html()).toContain('name="edit-button"');
-    expect(wrapper.html()).toContain('type="button"');
-  });
+    it('does not show the save button immediately', () => {
+        expect(wrapper.contains('#buttonSave')).toBe(false);
+    });
+    it('does not show the edit input immediately', () => {
+        expect(wrapper.contains('#todoTextEdited')).toBe(false);
+    });
 
-  it('doesn\'t show the edit input immediately', () => {
-    expect(wrapper.html()).not.toContain('<input type="text" name="newText">');
-  });
+    describe('layout', () => {
+        it('has rows', () => {
+            expect(wrapper.html()).toContain('<div class="row mx-lg-n5 jest-list-item">');
+        });
+        it('has columns', () => {
+            expect(wrapper.html()).toContain('<div class="col py-3 border bg-light">');
+        });
+    });
 
-  const editButton = wrapper.findAll('button').at(0);
-  const deleteButton = wrapper.findAll('button').at(1);
 
-  describe('when the edit button is pressed', () => {
-    editButton.trigger('click');
-    it('emits a ToggleEditMode event', () => {
-      expect(wrapper.emitted('toggleEditMode'));
+    describe('when the edit button is pressed', () => {
+        beforeEach(() => {
+            wrapper.find('#buttonEdit').trigger('click');
+            //wrapper.vm.$emit('toggleEditMode');
+            //wrapper.vm.$nextTick();
+        });
+        it('emits a ToggleEditMode event', () => {
+            expect(wrapper.emitted('toggleEditMode'));
+        });
+        // would be cool if they would work... however html does not change? maybe need to wait/update wrapper?
```

Comment:
I don't know. Sometimes it helps to `wait`, e.g. if you click on something that might create a promise. Then you would need to wait for it in your tests. Sometimes it's because of a timeout, which needs to be flushed: https://jestjs.io/docs/en/timer-mocks Sometimes it's because your assumptions regarding your own code are not correct.